src/pykx/pandas_api/pandas_meta.py

```diff
@@ -228,6 +228,21 @@ def max(self, axis=0, skipna=True, numeric_only=False):
             res
         ), cols)
 
+    @convert_result
+    def idxmax(self, axis=0, skipna=True, numeric_only=False):
+        res, cols = preparse_computations(self, axis, skipna, numeric_only)
+        maximums = self.max(axis, skipna, numeric_only)
+        num_str = '9h$' if numeric_only else ''
+        func_str = '(::)' if axis==0 else 'column_names'
+        if(numeric_only):
+            (_, column_names) = _get_numeric_only_subtable_with_bools(self)
+        else:
+            column_names = q('cols', self)
+        op= q("{[row;col;maximums;column_names]"
+              f"{func_str}[({num_str}row) ?' maximums[col]]"
+              "}")
+        return (op(res, cols, maximums, column_names), cols)
```

Comment:
This implementation looks a bit cumbersome. I would try to reimplement this in a more pythonic way. You can use `max` to find the maximum value of each list from `res` and then check its index with `.index`. For example, one way to do this would be:

```python
max_vals = [l.index(max(l)) for l in res]
```

Expanding this concept, we would have the following implementation:

```python
def idxmax(self, axis=0, skipna=True, numeric_only=False):
    tab = self
    res, cols = preparse_computations(self, axis, skipna, numeric_only)
    column_names = _get_numeric_only_subtable_with_bools(tab)[1] if numeric_only else tab.columns.py()
    max_vals = [l.index(max(l)) for l in res]

    return (max_vals if axis == 0 else column_names[max_vals], cols)
```

This probably still needs a bit of work (I haven't run the tests). I especially dislike the way we get the `column_names` variable, so feel free to iterate over it.

If you decide to iterate over your code and come up with a less pythonic implementation, there are a few things to comment on the specific q implementation of `op`, so feel free to respond to this comment and we can check them.